src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,166 @@
+package duke.main;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddTaskCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteTaskCommand;
+import duke.command.DisplayListCommand;
+import duke.command.DoneTaskCommand;
+import duke.command.ErrorCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.command.SaveCommand;
+import duke.error.DukeError;
+import duke.error.EmptyDateError;
+import duke.error.EmptyTaskError;
+import duke.error.InvalidDateInputError;
+import duke.error.InvalidNumberError;
+import duke.error.UnknownCommandError;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * The class that parses the user's input into commands that Duke can understand.
+ */
+public abstract class Parser {
+    /**
+     * Reads the user's input and returns a command for Duke to execute.
+     *
+     * @param inputLine The user's input
+     * @return Returns an executable command based on the user's input
+     */
+    public static Command parse(String inputLine) {
```

Comment:
The parse method look long and complicated, maybe you can try to refactor by extracting methods.
e.g. addDeadline method for case deadline